web/docs/guides/platform/performance.mdx

```diff
@@ -0,0 +1,70 @@
+| 2XL            | 380                    | 1500                  |
+| 4XL            | 480                    | 3000                  |
+| 8XL            | 490                    | 6000                  |
+| 12XL           | 500                    | 9000                  |
+| 16XL           | 500                    | 12000                 |
+
+If the number of connections is insufficient, you will receive the following error upon connecting to the DB:
+
+```shell
+$ psql -U postgres -h ...
+FATAL: remaining connection slots are reserved for non-replication superuser connections
+```
+
+In such a scenario, you can consider either upgrading to a larger compute add-on, or alternatively, by manually configuring the database for a higher number of connections.
+
+You can configure Postgresql by executing the following statement, followed by a server restart:
+
+```sql
+alter system set max_connections = '<val-here>';
+```
+
+Note that [the default configuration used by the Supabase platform](https://github.com/supabase/supabase-admin-api/blob/master/optimizations/postgres.go) optimizes the database to maximize resource utilization, and as a result, you might also need to configure other options (e.g. `work_mem`, `shared_buffers`, `maintenance_work_mem`) in order to tune them towards your use-case, and to avoid causing instability in your database.
+
+Furthermore, note that once overridden, the Supabase platform will continue to respect your manually configured value (even if the add-on size is changed), unless the override is removed with the following statement, followed by a server restart:
+
+```sql
+alter system reset max_connections;
+alter system reset <other-overridden-conf>;
+...
+```
+
+Configuring the number of PGBouncer connections is not supported at this time.
+
+## Examining Query Performance
+
+Each Supabase project comes with [the pg_stat_statements extension](https://www.postgresql.org/docs/14/pgstatstatements.html) enabled out of the box. This extension records query execution performance details that can be reviewed in order to find the queries that are e.g. taking the most time to execute, which you can then use in combination with the Postgresql query plan analyzer in order to come up with strategies for more efficient querying.
+
+Obtaining information from pg_stat_statements:
+```sql
+select mean_exec_time + stddev_exec_time, * from pg_stat_statements order by 1 desc;
+```
+
+Using the query plan analyzer on your expensive queries:
+
+```sql
+explain analyze <query-statement-here>;
+```
+
+Using the query plan analyzer to optimize your queries is a massive topic in itself, with [a number](https://www.postgresql.org/docs/current/using-explain.html) [of online](https://theartofpostgresql.com/explain-plan-visualizer/) [resources] (https://wiki.postgresql.org/wiki/Using_EXPLAIN) [available](https://www.enterprisedb.com/blog/postgresql-query-optimization-performance-tuning-with-explain-analyze).
```

Comment:
```suggestion
Be careful using `explain analyze` with `insert`/`update`/`delete` queries, because the query will actually run, and could have unintended side-effects.

Using the query plan analyzer to optimize your queries is a large topic, with a number of online resources available:

- [Official docs.](https://www.postgresql.org/docs/current/using-explain.html)
- [The Art of PostgreSQL.](https://theartofpostgresql.com/explain-plan-visualizer/)
- [Postgres Wiki.](https://wiki.postgresql.org/wiki/Using_EXPLAIN)
- [Enterprise DB.](https://www.enterprisedb.com/blog/postgresql-query-optimization-performance-tuning-with-explain-analyze)

```